### PR TITLE
Mooish Attribute Shortcuts For Everyone!

### LINKED ISSUE #334: Moose attribute shortcuts for everyone
Which Perl module do you want to write about: Mooish::AttributeBuilder

Please give a short description of the article you want to write. Two to three sentences is fine:
I've no clue how to turn it into a Christmas story, but since there is shortage of articles this year I'll happily write a short article about my Mooish::AttributeBuilder module. It's a no-dependency helper for a wide range of modules using Moose syntax, so I think it will be well received judging from my similar entry last year.

### PR DESCRIPTION
**Pull request by bbrtj**
I don't like the ending very much, so I'm open to suggestions.

Closes #334

### CODE REVIEW

**Comment by oalders** [comment]
@openstrike -- any suggestions for the ending?

**Comment by openstrike** [comment]
The ending's OK, if a little abrupt. I wouldn't worry about it.

The article talks about [MooseX::Extended](https://metacpan.org/pod/MooseX::Extended) but doesn't seem to mention whether or not you can use that and [Mooish::AttributeBuilder](https://metacpan.org/pod/Mooish::AttributeBuilder) together in the same module/script without issue. Given that they provide the same keywords that might confuse someone like me who rarely uses Moose. Maybe a sentence or two on that might tie it all together?

**Comment by bbrtj** [comment]
@openstrike fair point, I don't think there's any gain from Mooish::AttributeBuilder if you already use MooseX::Extended, but the article may not state that clearly. I'll change it later to be more clear about the purpose of the module.

**Comment by bbrtj** [comment]
@openstrike could you take a look again? Had more writing inspiration today, I think the article is much better now.

File: 2023/incoming/mooish-attribute-shortcuts-for-everyone-.pod [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+To solve this, you just install L<MooseX::Extended> and enjoy new C<param>
+and C<field>, right? Sort of... The module does a lot of things, has a lot of
+dependencies and only works with Moose. If you often use L<Moo> instead of
+Moose, you will have to find a Moo-specific module for this, and the API will
+likely be different.
+
+This is the exact issue solved by L<Mooish::AttributeBuilder>. It is a
+zero-dependency, module-agnostic helper for Moose family of modules letting you
+build your attributes easier and giving them meaning. It's not by any means a
```
Comment (openstrike) [inline 2023/incoming/mooish-attribute-shortcuts-for-everyone-.pod:24]:
```suggestion
build your attributes more easily and give them meaning. It's not by any means a
```

File: 2023/incoming/mooish-attribute-shortcuts-for-everyone-.pod [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+# becomes:
+
+has '+parent_attribute' => (
+  writer => '_set_parent_attribute',
+);
+
+=end perl
+
+The module delivers some additional features, the main one being custom
+shortcuts, but the core functionality alone should immensly improve the code
```
Comment (openstrike) [inline 2023/incoming/mooish-attribute-shortcuts-for-everyone-.pod:170]:
```suggestion
shortcuts, but the core functionality alone should immensely improve the code
```

**Review by openstrike — CHANGES_REQUESTED** [review]
Thanks - looks really good with the re-wording. Just a couple of typos introduced with this to fix and I think the article will be ready to go!

**Review by openstrike — APPROVED** [review]
Looks good to merge to me. Thanks.

**Comment by oalders** [comment]
Thanks @bbrtj and @openstrike!

**Comment by oalders** [comment]
This will be published for Dec 9. https://github.com/perladvent/Perl-Advent/pull/354/files

**Pull request merged**